docs/content/guides/dagster/branch_depl_with_snowflake.mdx

```diff
@@ -0,0 +1,325 @@
+
+### Overview
+
+We have a `PRODUCTION` Snowflake database with a schema named `HACKER_NEWS`. In our production cloud environment, we’d like to write tables to Snowflake containing subsets of Hacker News data. These tables will be:
+
+- `items` (the entire dataset)
+- `comments` (data about comments)
+- `stories` (data about stories)
+
+In order to set up a branch deployment workflow to construct and test these tables, we will:
+
+1. Define these tables as [assets](https://docs.dagster.io/concepts/assets/software-defined-assets).
+2. Configure our assets to write to Snowflake using a different connection (credentials and database name) for two environments: production and branch deployment.
+3. Clone our production database upon each branch deployment launch, naming each clone `PRODUCTION_CLONE_<ID>`, where `<ID>` is the pull request ID of the branch. We’ll create a branch deployment to test our Hacker News assets.
+4. Delete the corresponding database clone upon closing the feature branch.
+
+### Step 1: Create our job
+
+In production, we want to write three tables to Snowflake: items, comments, and stories. We can define these tables as [assets](https://docs.dagster.io/concepts/assets/software-defined-assets) as follows:
+
+```python file=/guides/dagster/local_to_prod_tutorial/assets.py startafter=start_assets endbefore=end_assets
+# assets.py
+
+
+@asset(
+    config_schema={"N": int},
+    required_resource_keys={"hn_client"},
+)
+def items(context) -> pd.DataFrame:
+    """Items from the Hacker News API: each is a story or a comment on a story."""
+    hn_client = context.resources.hn_client
+
+    rows = []
+    # Hacker News API is 1-indexed, so adjust range by 1
+    for item_id in range(context.op_config["N"] + 1, hn_client.fetch_max_item_id() + 1):
+        rows.append(hn_client.fetch_item_by_id(item_id))
+
+    result = pd.DataFrame(rows, columns=hn_client.item_field_names).drop_duplicates(
+        subset=["id"]
+    )
+    result.rename(columns={"by": "user_id"}, inplace=True)
+    return result
+
+
+@asset
+def comments(items: pd.DataFrame) -> pd.DataFrame:
+    """Comments from the Hacker News API."""
+    return items[items["type"] == "comment"]
+
+
+@asset
+def stories(items: pd.DataFrame) -> pd.DataFrame:
+    """Stories from the Hacker News API."""
+    return items[items["type"] == "story"]
+```
+
+As you can see, our assets use a [resource](https://docs.dagster.io/\_apidocs/libraries/dagster-snowflake#dagster_snowflake.snowflake_resource) named `hn_client` (source code available in the [Transitioning Data Pipelines from Development to Production](/guides/dagster/local-to-production-tutorial#part-one-local-development) guide). Using a resource allows us to swap out implementations per environment without modifying our computational code.
```

Comment:
I recently restructured the assets again to remove the hn client resource and only introduce it in the Testing section. so if you use the version without the hn resource this can be deleted

Reply:
"computational code" is a little redundant - maybe replace with "business logic"?